src/utils/isReactComponentClass.js

```diff
@@ -44,6 +44,27 @@ export default function isReactComponentClass(
     return true;
   }
 
+  // check for @extends React.Component in docblock
+  if (path.parentPath && path.parentPath.value) {
+    var classDeclaration;
+    if (Array.isArray(path.parentPath.value)) {
+      var matches = path.parentPath.value.filter(function(declaration) { return declaration.type === 'ClassDeclaration' });
+      if (matches[0]) {
+        classDeclaration = matches[0];
+      }
+    } else {
+      classDeclaration = path.parentPath.value;
+    }
+    
+    if (classDeclaration &&
```

Comment:
Could tighten this up a bit more:

``` js
if (!classDeclaration || !classDeclaration.leadingComments) {
  return false;
}

return classDeclaration.leadingComments.some(function (comment) {
  return /@extends\s+React\.Component/.test(comment.value);
});
```

Reply:
I don't think we could do this. 
```
if (!classDeclaration || !classDeclaration.leadingComments) {
  return false;
}
```
^ This would cause an early return. We still need to check to see if the component directly extends from `React.Component`.